jwst/pipeline/calwebb_tso3.py

```diff
@@ -0,0 +1,143 @@
+                        err=cube.err[i], dq=cube.dq[i])
+                image.meta = cube.meta
+                input_models.append(image)
+            
+            if not self.scale_detection:
+                self.log.info("Performing outlier detection on input images...")
+                results = self.outlier_detection(input_models)
+            else:
+                self.log.info("Performing scaled outlier detection on input images...")
+                results = self.outlier_detection_scaled(input_models)
+                
+            # Transfer updated DQ values to original input observation
+            for i in range(cube.data.shape[0]):
+                # preserve output filename 
+                orig_filename = cube.meta.filename
+                # Update DQ arrays with those from outlier_detection step
+                cube.dq[i] = input_models[i].dq
+                # reset output filename to original value
+                cube.meta.filename = orig_filename
+
+        self.log.info("Writing Level 2c images with updated DQ arrays...")
+        suffix_2c = 'crfints'
+        for cube in input_asn:
+            self.output_basename = cube.meta.filename
+            self.save_model(cube, suffix=suffix_2c)
+
+        # Create final photometry results as a single output
+        # regardless of how many members their may be...
+        phot_result_list = []
+        if 'image' in results[0].meta.exposure.type.lower():
+            # Create name for extracted photometry (Level 3) products
+            phot_tab_name = "{}_phot.ecsv".format(product_name)
+
+            # For each cube, extract photometry...
+            for cube in input_asn:
+                # Extract Photometry from imaging data
+                phot_result_list.append(self.tso_photometry(cube))
+        else:
+            # Working with spectroscopic TSO data...
+            # define output for x1d (level 2c/3) products
+            x1d_models = datamodels.ModelContainer()
+            x1d_models.update(input_asn)
+            #x1d_models.meta = input_asn.meta
+
+            # Create name for extracted white-light (Level 3) products
+            phot_tab_name = "{}_whtlt.ecsv".format(product_name)
+
+            # For each exposure in the TSO...
+            for cube in input_asn:
+                # Process spectroscopic TSO data
+                # extract 1D
+                self.log.info("Extracting 1-D spectra...")
+                self.extract_1d.suffix = 'x1dints'
+                result = self.extract_1d(cube)
+                x1d_models.append(result)
+                #
+                # perform white-light photometry on 1d extracted data
+                self.log.info("Performing white-light photometry...")
+                phot_result_list.append(self.white_light(result))
+                
+            # Define suffix for stack of extracted 1d (level 2b) products
+            #self.save_model(x1d_models,suffix="x1dints") # did not work!!!
```

Comment:
Discussing this with Howard, we realized that the code is trying to use the .save_model() method on a ModelContainer created within the Step.  This may be causing this code to not initialize everything as needed to work; the method simply runs without complaint and produces no file on disk.